File: Ch03/03_01/collatz.py
def collatz_step(n):
    """Compute one step in Collatz conjecture"""
    if n % 2 == 0:
        return n // 2
    return n * 3 + 1


if __name__ == '__main__':
    print(collatz_step(4))
    print(collatz_step(5))

File: Ch03/03_01/collatz.go
package main

import "fmt"

// CollatzStep computes one step in CollatzStep conjecture
func CollatzStep(n int) int {
	if n%2 == 0 {
		return n / 2
	}
	return n*3 + 1
}

func main() {
	fmt.Println(CollatzStep(4))
	fmt.Println(CollatzStep(5))
}
